akka-http-tests/src/test/scala/akka/http/scaladsl/server/TestServer.scala

```diff
@@ -21,18 +25,35 @@ object TestServer extends App {
     akka.log-dead-letters = off
     akka.stream.materializer.debug.fuzzing-mode = off
     """)
+
   implicit val system = ActorSystem("ServerTest", testConf)
   import system.dispatcher
   implicit val materializer = ActorMaterializer()
 
+  // --------- json streaming ---------
+  import spray.json.DefaultJsonProtocol._
+  import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+  final case class Tweet(message: String)
+  implicit val tweetFormat = jsonFormat1(Tweet)
+
+  // FIXME: Need to be able to support composive framing with content type (!!!!!!!)
+  import akka.http.scaladsl.server.EntityStreamingSupport._
+  /* override if extending EntityStreamingSupport */
+  implicit val incomingEntityStreamFraming: FramingWithContentType = bracketCountingJsonFraming(128)
+  /* override if extending EntityStreamingSupport */
+  implicit val outgoingEntityStreamRendering: SourceRenderingMode = JsonSourceRenderingModes.LineByLine
```

Comment:
Currently one has to know that `EntityStreamingSupport.bracketCountingJsonFraming` has to be used for "reading" and `JsonSourceRenderingModes.LineByLine` for "writing" which isn't too obvious IMO. I'm just trying to streamline that a bit.

I think the most imported aspect of this PR should be to "get the API right". Other things could be improved later on. Sorry for coming up with this stuff only now.